cohere-cookbooks/Semantic search with Cohere embeddings/Wikipedia_Semantic_Search_With_Cohere_Embeddings_Archives.ipynb

```diff
@@ -0,0 +1,384 @@
+      "https://simple.wikipedia.org/wiki/Alan%20Turing \n",
+      "\n",
+      "Alan Turing\n",
+      "In 2013, almost 60 years later, Turing received a posthumous Royal Pardon from Queen Elizabeth II. Today, the “Turing law” grants an automatic pardon to men who died before the law came into force, making it possible for living convicted gay men to seek pardons for offences now no longer on the statute book.\n",
+      "https://simple.wikipedia.org/wiki/Alan%20Turing \n",
+      "\n",
+      "Botany\n",
+      "Gregor Mendel (1822–1884), Augustinian priest and scientist, and is often called the father of genetics for his study of the inheritance of traits in pea plants.\n",
+      "https://simple.wikipedia.org/wiki/Botany \n",
+      "\n",
+      "Creativity\n",
+      "Creativity is the ability of a person or group to make something new and useful or valuable, or the process of making something new and useful or valuable. It happens in all areas of life - science, art, literature and music.\n",
+      "https://simple.wikipedia.org/wiki/Creativity \n",
+      "\n",
+      "Computer\n",
+      "In 1837, Charles Babbage proposed the first general mechanical computer, the Analytical Engine. The Analytical Engine contained an Arithmetic Logic Unit, basic flow control, punched cards, and integrated memory. It is the first general-purpose computer concept that could be used for many things and not only one particular computation. However, this computer was never built while Charles Babbage was alive, because he didn't have enough money. In 1910, Henry Babbage, Charles Babbage's youngest son, was able to complete a portion of this machine and perform basic calculations.\n",
+      "https://simple.wikipedia.org/wiki/Computer \n",
+      "\n",
+      "Alan Turing\n",
+      "Using cryptanalysis, he helped to break the codes of the Enigma machine. After that, he worked on other German codes.\n",
+      "https://simple.wikipedia.org/wiki/Alan%20Turing \n",
+      "\n",
+      "Angel\n",
+      "The same cherubim creatures were said to be cast in gold on top of the Ark of the Covenant. Casting metal is one of the oldest forms of artwork, and was attempted by Leonardo da Vinci.\n",
+      "https://simple.wikipedia.org/wiki/Angel \n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Compute dot score between query embedding and document embeddings\n",
+    "dot_scores = torch.mm(query_emb_int8, doc_embeddings.transpose(0, 1))\n",
+    "top_k = torch.topk(dot_scores, k)\n",
+    "\n",
+    "# Print results\n",
+    "print(\"Query:\", query)\n",
+    "for doc_id in top_k.indices[0].tolist():\n",
+    "    print(docs[doc_id]['title'])\n",
+    "    print(docs[doc_id]['text'])\n",
+    "    print(docs[doc_id]['url'], \"\\n\")"
+   ]
+  },
+  {
+   "cell_type": "code",
```

Comment:
Also, i'll leave this up to you as well as letting others comment in here. But I was expecting to see how the model could use the results of the semantic search to augment the response. Maybe, you can add a new section after the code snippet using the TOP K results from the semantic search as part of the prompt to then visualize how the model sticks to the provided data to get up to date results.

Reply:
Added a new section after the code snippet to explain how the TOP K results from the semantic search are useful when creating a prompt with context.